### PR TITLE
Fix #3286: maintain the SoundButton mute state in screen rotation scenario

### LINKED ISSUE #3286: VoiceInstruction's mute state and icon state doesn't match after screen rotation.
**Android API:** Any
**Mapbox Navigation SDK version:** Navigation UI SDK 1.0

### Steps to trigger behavior

 1. Launch **NavigationViewActivity** in the `example` test app
 2. Mute the VoiceInstruction
 3. Rotate screen
 4. Unmute the VoiceInstruction

### Expected behavior
The voice instruction can be played.

### Actual behavior
The voice instruction is still muted.

ezgif com-video-to-gif

### PR DESCRIPTION
**Pull request by JunDai**
## Description

Fix #3286: maintain the SoundButton mute state in screen rotation scenario

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Implementation

Need to restore the mute state in `NavigationViewModel` and `SoundButton`. Not only the icon resource.

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR
- [ ] We might need to update / push `api/current.txt` files after running `$> make 1.0-core-update-api` (Core) / `$> make 1.0-ui-update-api` (UI) if there are changes / errors we're 🆗 with (e.g. `AddedMethod` changes are marked as errors but don't break SemVer) 🚀 If there are SemVer breaking changes add the `SEMVER` label. See [Metalava](https://github.com/mapbox/mapbox-navigation-android/blob/master/docs/metalava.md) docs

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt [hunk 1]
```diff
@@ -101,7 +101,7 @@ class NavigationViewActivity : AppCompatActivity(), OnNavigationReadyCallback,
                 optionsBuilder.directionsRoute(route)
                 optionsBuilder.shouldSimulateRoute(true)
                 optionsBuilder.bannerInstructionsListener(this)
-                optionsBuilder.muteVoiceGuidance(true)
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt:104]: Should we keep it to showcase how to use the `muteVoiceGuidance` API?

**Comment by JunDai** [comment]
@Guardiola31337 - thanks for reporting the issue #3290 
This PR also updates to fix it.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewOptions.java [hunk 2]
```diff
@@ -186,6 +186,7 @@ public static Builder builder(Context context) {
         .roundingIncrement(Rounding.INCREMENT_FIFTY)
         .shouldSimulateRoute(false)
         .waynameChipEnabled(true)
+        .muteVoiceGuidance(false)
```
Comment (JunDai) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewOptions.java:189]: Fix #3290 @Guardiola31337

File: examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt (hunk 1 above)
Comment (JunDai) [reply examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt:104]: yeah, I was thinking about this. But I am afraid the developer will be confusing about why they can't hear the voice instruction.

File: examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewFragment.kt [hunk 3]
```diff
@@ -107,6 +107,7 @@ class NavigationViewFragment : Fragment(), OnNavigationReadyCallback, Navigation
                 optionsBuilder.directionsRoute(route)
                 optionsBuilder.shouldSimulateRoute(true)
                 optionsBuilder.enableVanishingRouteLine(true)
+                optionsBuilder.muteVoiceGuidance(true)
```
Comment (JunDai) [inline examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewFragment.kt:110]: https://github.com/mapbox/mapbox-navigation-android/pull/3289/files#r450393546

considering this example might be more suitable.

**Pull request merged**